Source/Scene/Cesium3DTileBatchTable.js

```diff
@@ -0,0 +1,1519 @@
+        var componentCount = binaryProperty.componentCount;
+        if (componentCount === 1) {
+            return typedArray[index];
+        }
+        return binaryProperty.type.unpack(typedArray, index * componentCount);
+    }
+
+    function setBinaryProperty(binaryProperty, index, value) {
+        var typedArray = binaryProperty.typedArray;
+        var componentCount = binaryProperty.componentCount;
+        if (componentCount === 1) {
+            typedArray[index] = value;
+        } else {
+            binaryProperty.type.pack(value, typedArray, index * componentCount);
+        }
+    }
+
+    // The size of this array equals the maximum instance count among all loaded tiles, which has the potential to be large.
+    var scratchVisited = [];
+    var scratchStack = [];
+    var marker = 0;
+    function traverseHierarchyMultipleParents(hierarchy, instanceIndex, endConditionCallback) {
+        var classIds = hierarchy.classIds;
+        var parentCounts = hierarchy.parentCounts;
+        var parentIds = hierarchy.parentIds;
+        var parentIndexes = hierarchy.parentIndexes;
+        var instancesLength = classIds.length;
+
+        // Ignore instances that have already been visited. This occurs in diamond inheritance situations.
+        // Use a marker value to indicate that an instance has been visited, which increments with each run.
+        // This is more efficient than clearing the visited array every time.
+        var visited = scratchVisited;
+        visited.length = Math.max(visited.length, instancesLength);
+        var visitedMarker = ++marker;
+
+        var stack = scratchStack;
+        stack.length = 0;
+        stack.push(instanceIndex);
+
+        while (stack.length > 0) {
+            instanceIndex = stack.pop();
+            if (visited[instanceIndex] === visitedMarker) {
+                // This instance has already been visited, stop traversal
+                continue;
+            }
+            visited[instanceIndex] = visitedMarker;
+            var result = endConditionCallback(hierarchy, instanceIndex);
+            if (defined(result)) {
+                // The end condition was met, stop the traversal and return the result
+                return result;
+            }
+            var parentCount = parentCounts[instanceIndex];
+            var parentIndex = parentIndexes[instanceIndex];
+            for (var i = 0; i < parentCount; ++i) {
+                var parentId = parentIds[parentIndex + i];
+                // Stop the traversal when the instance has no parent (its parentId equals itself)
+                // else add the parent to the stack to continue the traversal.
+                if (parentId !== instanceIndex) {
+                    stack.push(parentId);
+                }
+            }
+        }
+    }
+
+    function traverseHierarchySingleParent(hierarchy, instanceIndex, endConditionCallback) {
+        while (true) { // eslint-disable-line no-constant-condition
```

Comment:
I know I mentioned this in other pulls, so this is the last time I'll bring it up, but let's get rid of `while(true)`